### PR TITLE
[VL] Daily Update Velox Version (2024_04_12)

### PR DESCRIPTION
**Pull request by GlutenPerfBot**
Upstream Velox's New Commits:

```txt
0643556a7 by duanmeng, Add pool name regex in TestScopedSpillInjection (9413)
efb0213a4 by Masha Basmanova, Optimize cast(JSON as ROW) (9449)
707dbfddf by Masha Basmanova, Fix CAST(JSON as ROW(ARRAY)) (9447)
d5f8e8e7d by zhli1142015, Store duplicate row address in vector for join probe (9079)
f51e34d45 by Jacob Wujciak-Jens, Add missing `working-directory` in scheduled.yml (9454)
719d95678 by Jacob Wujciak-Jens, Fix syntax error in workflow yaml (9451)
d4f8d8556 by wypb, Fix the incorrect size calculation logic of FileSink. (9429)
8e9496bfb by Terry Wang, Fix a typo in the documentation about how to create scalar constant vector (9428)
2a33e63dd by Ankita Victor, Initialize FileOptions::fileSize in S3FileSystem::openFileForWrite (9441)
cef85a195 by Jacob Wujciak-Jens, Disable signature related functions in scheduled fuzzer runs (9420)
ecf811833 by Sergey Pershin, Make default value for "max_spill_level" equal to 1. (9419)
1c7de0122 by Daniel Munoz, Fix TSAN data race flaky test (9393)
acd204feb by Wei He, Fix approx_distinct result verifier for window fuzzer (9409)
```

diff https://github.com/oap-project/velox/compare/387ae3b9c1be14ed907d14dbe484a35c5357a32b..0643556a7

### CODE REVIEW

**Comment by zhztheplayer** [comment]
Failure related to https://github.com/facebookincubator/velox/commit/ecf811833da61d951e60f927f745b78969f27213

File: shims/common/src/main/scala/org/apache/gluten/GlutenConfig.scala [hunk 1]
```diff
@@ -1249,6 +1251,13 @@ object GlutenConfig {
       .checkValues(Set("none", "auto"))
       .createWithDefault("auto")
 
+  val COLUMNAR_VELOX_MAX_SPILL_LEVEL =
+    buildConf("spark.gluten.sql.columnar.backend.velox.maxSpillLevel")
+      .internal()
+      .doc("The max allowed spilling level with zero being the initial spilling level")
+      .intConf
+      .createWithDefault(6)
```
Comment (zhouyuan) [inline shims/common/src/main/scala/org/apache/gluten/GlutenConfig.scala:1259]: do we really allow this to be set to 0?

Comment (zhztheplayer) [reply shims/common/src/main/scala/org/apache/gluten/GlutenConfig.scala:1259]: It seems that a 0 indicates that spill is disabled. It's OK to allow the setting. (0 is the initial spilling level)

File: .github/workflows/velox_docker.yml [hunk 2]
```diff
@@ -245,7 +245,8 @@ jobs:
             -d=OFFHEAP_SIZE:6g,spark.memory.offHeap.size=6g \
             -d=OFFHEAP_SIZE:4g,spark.memory.offHeap.size=4g \
             -d=OVER_ACQUIRE:0.3,spark.gluten.memory.overAcquiredMemoryRatio=0.3 \
-            -d=OVER_ACQUIRE:0.5,spark.gluten.memory.overAcquiredMemoryRatio=0.5
+            -d=OVER_ACQUIRE:0.5,spark.gluten.memory.overAcquiredMemoryRatio=0.5 \
+            --excluded-dims=OFFHEAP_SIZE:4g,OVER_ACQUIRE:0.5
```
Comment (zhztheplayer) [inline .github/workflows/velox_docker.yml:249]: This case is failing. Should find out the reason later.

**Comment by zhztheplayer** [comment]
Failure may be caused by https://github.com/apache/incubator-gluten/pull/5377

**Pull request merged**